dist/pythonlibs/riotctrl_shell/netif.py

```diff
@@ -0,0 +1,367 @@
+            return True
+        return False
+
+
+class NetifStatsParser(ShellInteractionParser):
+    def __init__(self):
+        self.header_c = re.compile(r"Statistics for (?P<module>.+)$")
+        self.rx_c = re.compile(r"RX packets\s+(?P<packets>\d+)\s+"
+                               r"bytes\s+(?P<bytes>\d+)$")
+        self.tx_c = re.compile(r"TX packets\s+(?P<packets>\d+)\s+"
+                               r"\(Multicast:\s+(?P<multicast>\d+)\)\s+"
+                               r"bytes\s+(?P<bytes>\d+)$")
+        self.tx_err_c = re.compile(r"TX succeeded\s+(?P<succeeded>\d+)\s+"
+                                   r"errors\s+(?P<errors>\d+)$")
+
+    def parse(self, cmd_output):
+        """
+        Parses output of Netif::netif_stats or the statistics part of
+        Netif::netif_list.
+
+        :param cmd_output(str): output of Netif::netif_stats or
+                                Netif::netif_list
+        :return: dictionary with one entry per statistics module, each module
+                 containing an entry 'rx' and 'tx' with respective statistics
+
+        >>> parser = NetifStatsParser()
+        >>> res = parser.parse(
+        ...         "Statistics for IPv6\\n"
+        ...         "  RX packets 14  bytes 1104\\n"
+        ...         "  TX packets 3 (Multicast: 1)  bytes 192\\n"
+        ...         "  TX succeeded 3 errors 0\\n")
+        >>> sorted(res)
+        ['IPv6']
+        >>> sorted(res["IPv6"])
+        ['rx', 'tx']
+        >>> sorted(res["IPv6"]["rx"])
+        ['bytes', 'packets']
+        >>> sorted(res["IPv6"]["tx"])
+        ['bytes', 'errors', 'multicast', 'packets', 'succeeded']
+        >>> res["IPv6"]["rx"]["bytes"]
+        1104
+        """
+        stats = None
+        current = None
+        for line in cmd_output.splitlines():
+            m = self.header_c.search(line)
+            if m is not None:
+                module = m.group("module")
+                if stats is None:
+                    stats = {}
+                current = stats[module] = {}
+            if current is not None:
+                if "rx" not in current:
+                    m = self.rx_c.search(line)
+                    if m is not None:
+                        current["rx"] = {k: int(v)
+                                         for k, v in m.groupdict().items()}
+                elif "tx" not in current:
+                    m = self.tx_c.search(line)
+                    if m is not None:
+                        current["tx"] = {k: int(v)
+                                         for k, v in m.groupdict().items()}
+                elif "tx" in current:
+                    m = self.tx_err_c.search(line)
+                    if m is not None:
+                        current["tx"].update(
+                            {k: int(v) for k, v in m.groupdict().items()}
+                        )
+        return stats
+
+
+# ==== ShellInteractions ====
+
+class Netif(ShellInteraction):
```

Comment:
How about:

```suggestion
class Ifconfig(ShellInteraction):
```
?

Since this is a wrapper around the ifconfig shell command.

Reply:
Done.